### PR TITLE
fix(react): handle replace actions better, closes #21832

### LINKED ISSUE #21832: bug: Navigation: chain of push/replace
# Bug Report

**Ionic version:**

[ ] **4.x**
[x] **5.x**

**Steps to reproduce:**
`https://github.com/Elardzhi/Ionic-React-navigation.git`

**Description:**
In the demo repository, we have a chain of `history.replace` and `history.push`.
Page 1 -> Page 2 : history.replace, no animation, working as expected
Page 2 -> Page 3 : history.push, transition with animation, working as expected
Page 3 -> Page 4 : history.replace, **animation exists, not expected**
Page 4 -> Page 1 : history.replace, **animation exists, not expected**
Page 1 -> Page 2 : history.replace, URL is updated, but **no transition to Page 2**

### PR DESCRIPTION
**Pull request by elylucas**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: packages/react-router/src/ReactRouter/IonRouter.tsx [hunk 1]
```diff
@@ -173,13 +172,15 @@ class IonRouterInner extends React.PureComponent<IonRouteProps, IonRouteState> {
           const currentRouteInfo = this.locationHistory.current();
           routeInfo.lastPathname = currentRouteInfo?.pathname || routeInfo.lastPathname;
           routeInfo.pushedByRoute = currentRouteInfo?.pushedByRoute || routeInfo.pushedByRoute;
-          routeInfo.routeDirection = currentRouteInfo?.routeDirection || routeInfo.routeDirection;
-          routeInfo.routeAnimation = currentRouteInfo?.routeAnimation || routeInfo.routeAnimation;
```
Comment (liamdebeasi) [inline packages/react-router/src/ReactRouter/IonRouter.tsx:177]: Do custom `routeAnimation` props still work here?

Comment (elylucas) [reply packages/react-router/src/ReactRouter/IonRouter.tsx:177]: ya, potentially. This is what was causing the error though. I asked a question on the bug issue around the use case for doing navigations with replaces. I might put this back in based on the response to my question.

**Review by Butuhpakagsial — APPROVED** [review]
packages/react-router/src/ReactRouter/IonRouter.tsx

**Comment by sean-perkins** [comment]
Thanks for the PR Ely!

Based on the age of the PR, conflicts with main and no linked issue, we've decided to close out this pull request in the interim. The team will open a new pull request if an issue is reported in the future.

Thanks!

**Pull request closed without merging**